src/routes/insights/Insights.tsx

```diff
@@ -1,16 +1,23 @@
-import { Hammer } from '@phosphor-icons/react';
 import classes from './Insights.module.scss';
+import DistributionChart from './DistributionChart';
+import OverviewStats from './OverviewStats';
+import { useContext } from 'react';
+import { ActivitiesContext } from '../../data/contexts/ActivitiesContext';
+import { CategoriesContext } from '../../data/contexts/CategoriesContext';
 
 const Insights = () => {
-    return (
-        <div className={classes.insights}>
-            <h1 className={classes.header}>Insights</h1>
-            <div className={classes.content}>
-                <h2>Work in progress</h2>
-                <Hammer size={48} />
-            </div>
-        </div>
-    );
+  const activities = useContext(ActivitiesContext);
+  const { categories } = useContext(CategoriesContext);
+
+  return (
+    <div className={classes.insights}>
+      <h1 className={classes.header}>Insights</h1>
+      <div className={classes.content}>
+        <OverviewStats activities={activities} categories={categories} />
+        <DistributionChart activities={activities} categories={categories} />
```

Comment:
Note that you're still prop-drilling here.
What I meant for you to do is define `InsightsContext` as `{ activities, categories }`, wrap this component's JSX in `InsightsContext.provider` and in the children call `useContext` instead of passing props.
I'd rather avoid prop drilling as it'll become a burden the more we add charts into this page.